src/components/Section/History/dateranges.js

```diff
@@ -3,15 +3,18 @@
  * Date object or null.
  */
 export const extractDate = dateObj => {
+  
   if (dateObj instanceof Date) {
     return dateObj
   }
 
   if (!dateObj || !dateObj.month || !dateObj.day || !dateObj.year) {
     return null
   }
+  var d = new Date(`${dateObj.month}/${dateObj.day}/${dateObj.year}`)
```

Comment:
Is there a reason why this is changed? My understanding was that this PR is for the applicant's birthdate, but this change has to do with something else. Can you please clarify what this fixes?

Reply:
this fixes an issue with two digit years, so, if I were to enter 99 into the year textbox, it would validate since the date year will be 1999, it gets kinda screwy to the user I think, so this will make it the year 0099, in some cases like 15 becomes 2015 instead of 1915 so I think it's just confusing all over, I think this addresses that

Reply:
@leonged I'm just skimming this PR, but this doesn't sound like the type of change that should be done in this PR. A separate issue should be made, someone thinking about the product/usability should consider whether it is a good change to make, etc.

I can almost guarantee that if eApp automatically changes `99` to `1999` someone spent some time thinking about it, possibly even doing usability tests with users.

Reply:
I don't think eApp is doing it intentionally, I think that's how JavaScript date objects are... so previously it would give a date too far into the past error when something < 1000 was entered for year... so now that we decided it should just give a "age too old error", people can enter 99 and expect to get an age too old error, but that is not the case if we don't make this change since JavaScript dateobject will be 1999

Reply:
I don't think this should be included in this PR, especially since it hasn't been addressed with usability testing or required by the PO. If we want to add this kind of feature to eApp, it needs to be consistently addressed throughout the application.

I also don't think it works as intended. Maybe I'm executing it wrong..
// const d = new Date()
// d.setFullYear('01')
// Wed Sep 19 0001 08:36:08 GMT-0752
// d.setFullYear('99')
// Sat Sep 19 0099 08:36:08 GMT-0752

I think we're getting close enough to `2020` where `20` can mean '0020', `1920` or `2020`. `1920` can represent a birth date for a relative (mother or father, for example) OR it can represent the year `2020`, when someone get married recently (in the near future). From the example above, i suspect entering `20` will mean `0020`, but I don't know the exactly JS spec.

Reply:
That is how it's intended. Do you have a minute to chat?

Reply:
Okay @ryanhofdotgov. I just caught up with @leonged and I misunderstood the intent of that code. I'll continue to test it and see what comes out of it.

When a user enters the month as `10`, day as `21`, and year as `99`, we expect the year to be `0099`. When Javascript creates that date with `99`, it converts `99` to `1999`. So in this PR the `d.setFullYear('99')` sets it as `0099`.

The reason why he did this was to trigger the more specific error message (age must be 16-100; you're too old), rather than the less specific (this year is too far in the past)

Reply:
However, I think going forward, it might be worth thinking about or looking into enforcing date years to have 4 digits. That's not happening now.

Reply:
I think that's good, we can create a ticket to have the option of enforcing a length on the numbers, I don't know if that exists currently. We could also update the text to say enter DOB in mm/dd/yyyy format.